src/mlpack/methods/ann/layer/multi_layer_impl.hpp

```diff
@@ -248,8 +248,9 @@ void MultiLayer<MatType>::SetWeights(typename MatType::elem_type* weightsPtr)
     Log::Assert(start + weightSize <= totalWeightSize,
         "FNN::SetLayerMemory(): parameter size does not match total layer "
         "weight size!");
-
-    network[i]->SetWeights(weightsPtr + start);
+    MatType tmpWeights;
+    MakeAlias(tmpWeights, weightsIn, weightsIn.n_rows, weightsIn.n_cols, start);
```

Comment:
I think the size here is wrong, I suspect it should be `weightSize` x `1` or similar.  (I think the code still works as-is, but it's probably better to fix it so that the alias we pass to `network[i]` has size equal to how many weights `network[i]` will use.)

Reply:
To clarify what I mean is that the weights for the layer `network[i]` are not of size `weightsIn.n_rows x weightsIn.n_cols`---`weightsIn` is the matrix that represents *all* the weights for all the layers in `network`.  For layer `i` specifically, the size of the weights for that layer is `weightSize`, with offset `start`.  So, the matrix `tmpWeights` is too large.  If you change it to `MakeAlias(tmpWeights, weightsIn, weightSize, 1, start)`, I think it will be fine.